### PR TITLE
fix: Squashed minor search api bugs

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

File: api/server/v1/marshaler.go [hunk 1]
```diff
@@ -441,6 +441,36 @@ func (x *SearchHit) MarshalJSON() ([]byte, error) {
 	return json.Marshal(resp)
 }
 
+func (x *SearchMetadata) MarshalJSON() ([]byte, error) {
+	resp := struct {
+		Found      int64 `json:"found"`
```
Comment (adilansari) [inline api/server/v1/marshaler.go:446]: Found could be 0, hence, not omitting

File: server/services/v1/query_runner.go [hunk 2]
```diff
@@ -537,6 +537,7 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
 	}
 
 	facets, err := qsearch.UnmarshalFacet(runner.req.Facet)
+	// test if facet field is actually a schema field
```
Comment (adilansari) [inline server/services/v1/query_runner.go:540]: Will include this in another iteration, wanted to get server in sync with latest API

File: server/services/v1/search_reader.go [hunk 3]
```diff
@@ -18,16 +18,17 @@ import (
 	"context"
 
 	jsoniter "github.com/json-iterator/go"
+	tsApi "github.com/typesense/typesense-go/typesense/api"
+
```
Comment (himank) [inline server/services/v1/search_reader.go:22]: remove the line and run `go imports`.

File: store/search/ts.go [hunk 4]
```diff
@@ -16,15 +16,19 @@ package search
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"io"
 	"net/http"
 
+	"github.com/rs/zerolog/log"
+
```
Comment (himank) [inline store/search/ts.go:25]: ^

File: test/docker/docker-compose.yml [hunk 5]
```diff
@@ -86,7 +86,6 @@ services:
       # instead of test/config/fdb.cluster file when running on the host
       - TIGRIS_ENVIRONMENT=test
       - TEST_PARAM
-    container_name: tigris_test
```
Comment (himank) [inline test/docker/docker-compose.yml:89]: is this intentional?

Comment (adilansari) [reply test/docker/docker-compose.yml:89]: Already set above

File: server/services/v1/query_runner.go [hunk 6]
```diff
@@ -591,17 +580,31 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
 			})
-			totalInPage++
 
 			if len(resp.Hits) == pageSize {
 				break
 			}
 		}
-		if len(resp.Hits) == 0 {
+		if rowReader.err != nil {
+			return nil, ctx, rowReader.err
```
Comment (himank) [inline server/services/v1/query_runner.go:589]: We may have buffered the events in the `Hits` if we return an error from here then we won't be able to send those hits?

File: server/services/v1/query_runner.go [hunk 7]
```diff
@@ -591,17 +580,31 @@ func (runner *SearchQueryRunner) Run(ctx context.Context, tx transaction.Tx, ten
...
+			Found:      rowReader.getTotalFound(),
+			TotalPages: totalPages,
+			Page: &api.Page{
+				Current: pageNo,
+				Size:    int32(searchQ.PageSize),
+			},
+		}
+
+		// break the stream after sending at least 1 response
+		if len(resp.Hits) == 0 && pageNo != int32(defaultPageNo) {
```
Comment (himank) [inline server/services/v1/query_runner.go:604]: Shouldn't the condition here be OR and also what if the user passed the pageNo as `1` then also we need to stop?

File: server/services/v1/search_reader.go (hunk 3 above)
Comment (adilansari) [reply server/services/v1/search_reader.go:22]: apparently this is after goimports, it is grouping imports by package. Is there a particular configuration that we follow for goimports?

Comment (himank) [reply server/services/v1/search_reader.go:22]: So we remove the extra lines and then run `go imports`. This way there are only two groups of imports.

**Pull request closed without merging**